frontend/src/pages/create_campaign/index.js

```diff
@@ -0,0 +1,167 @@
+import Dropzone from 'react-dropzone';
+import { LoadingIndicator } from "../../components";
+import { fileToDataUrl } from "../../utils";
+
+const CreateCampaign = () => {
+  const navigate = useNavigate();
+
+  const [name, setName] = React.useState("");
+  const [startDate, setStartDate] = React.useState(new Date());
+  const [endDate, setEndDate] = React.useState(new Date());
+  const [description, setDescription] = React.useState("");
+  const [interviewStage, setInterviewStage] = React.useState(false);
+  const [scoringState, setScoringStage] = React.useState(false);
+  const [draft, setDraft] = React.useState(false);
+  const [cover, setCover] = React.useState(null);
+  const [error, setError] = React.useState(null);
+
+  // TODO: user authentication and redirection if they are not logged in or authenticated
+
+  const submitHandler = async () => {
+    if (name.length === 0 && !draft) {
+      setError("Campaign name is required");
+    } else if (description === 0 && !draft) {
+      setError("Campaign description is required");
+    } else if (startDate.getTime() > endDate.getTime()) {
+      setError("Start date must be before end date");
+    } else {
+      setError(null);
+    }
+
+    const coverSend = cover ? cover.slice(cover.indexOf(";base64,") + 8) : "";
+    const startTimeString = `${startDate.getFullYear()}-${startDate.getMonth() < 9 ? `0${startDate.getMonth() + 1}` : `${startDate.getMonth() + 1}`}-${startDate.getDate() < 9 ? `0${startDate.getDate() + 1}` : `${startDate.getDate() + 1}`}T${startDate.getHours() < 9 ? `0${startDate.getHours()}` : `${startDate.getHours()}`}:${startDate.getMinutes() < 9 ? `0${startDate.getMinutes()}` : `${startDate.getMinutes()}`}:00`;
+    const endTimeString = `${endDate.getFullYear()}-${endDate.getMonth() < 9 ? `0${endDate.getMonth() + 1}` : `${endDate.getMonth() + 1}`}-${endDate.getDate() < 9 ? `0${endDate.getDate() + 1}` : `${endDate.getDate() + 1}`}T${endDate.getHours() < 9 ? `0${endDate.getHours()}` : `${endDate.getHours()}`}:${endDate.getMinutes() < 9 ? `0${endDate.getMinutes()}` : `${endDate.getMinutes()}`}:00`;
+
+    const postCampaign = await fetch("http://127.0.0.1:8000/campaign/new", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        // TODO: replace organisation id with something in the frontend that returns the id, would necessitate an endpoint to get all orgs the  user is a part of
+        organisation_id: 1,
+        name,
+        description,
+        starts_at: startTimeString,
+        ends_at: endTimeString,
+        draft,
+        // draft now means that it is an actual draft
+        cover_image: coverSend,
+      }),
+    });
+
+    const status = await postCampaign.status;
+    if (status === 200) {
+      console.log("nice!");
+    } else {
+      console.log("something fucked up");
+    }
```

Comment:
Feel free to leave this as is - but we should add a snackbar/toastbar at some point